src/wallet/wallet.cpp

```diff
@@ -1054,6 +1056,7 @@ bool CWallet::AddToWalletIfInvolvingMe(const CTransactionRef& ptx, const CBlockI
                     std::map<CKeyID, int64_t>::const_iterator mi = m_pool_key_to_index.find(keyid);
                     if (mi != m_pool_key_to_index.end()) {
                         LogPrintf("%s: Detected a used keypool key, mark all keypool key up to this key as used\n", __func__);
+                        AddRelatedScripts(txout.scriptPubKey, mi->first);
```

Comment:
This is not sufficient? What if you have two keys in your pool, in order, A and B, then B gets used as a segwit address on the chain first, and then A...you'd only add B to your watch scripts?

Reply:
The keystore implicitly knows about all keys' corresponding redeemscripts. This function call is not needed except for downgrading to old software after recovery.

Reply:
Discussed this IRL. This was indeed a problem. It's fixed now by moving the call to `MarkReserveKeyAsUsed`.